pkgs/by-name/pr/proton-ge-custom/package.nix

```
@@ -0,0 +1,41 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, writeScript
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
```

If all you need is one phase to run a few commands, it's better to use

```suggestion
stdenvNoCC.runCommand (finalAttrs: {
```

---

I am not sure how to convert this derivation to runCommand without losing some of the fine-graining that we have. Could you give me some pointers for the conversion?

p.s. is stdenvNoCC.runCommand a thing? Is it not pkgs.runCommand?

---

WDYM by "fin-graining"?

All this does is declare a few commands AFAICT.

---

To my knowledge runCommand does not have meta and updatescript.

---

runCommand is a drv. There should be nothing preventing you from giving it a meta or passthru fields.

---

Ah, one downside of runCommand is that it doesn't really support `pname + version`; you must give it `name`.

In that case, `stdenvNoCC.mkDerivation` with a custom buildPhase actually sounds like a pretty sensible option; especially the NoCC part as it disable stripping etc.